### PR TITLE
txn: attach cluster id in kv request

### PR DESCRIPTION
**Pull request by cfzjywxk**
Ref: https://github.com/tikv/tikv/issues/16545

Adding cluster id information into the kv requests.

### CODE REVIEW

File: internal/locate/region_cache.go [hunk 1]
```diff
@@ -671,6 +672,12 @@ func NewRegionCache(pdClient pd.Client) *RegionCache {
 	c.notifyCheckCh = make(chan struct{}, 1)
 	c.bg = newBackgroundRunner(context.Background())
 	c.enableForwarding = config.GetGlobalConfig().EnableForwarding
+	if c.pdClient != nil {
+		c.clusterID = c.pdClient.GetClusterID(context.Background())
+	}
+	if c.clusterID == 0 {
+		logutil.BgLogger().Error("cluster id is not set properly")
+	}
```
Comment (zyguan) [inline internal/locate/region_cache.go:680]: `RegionCache` exports a method `SetPDClient`, maybe we can move the code of cluster id initialization there.

Comment (cfzjywxk) [reply internal/locate/region_cache.go:680]: The `SetPDClient` is marked as `for test only` in the comments.

**Review by zyguan — APPROVED** [review]

**Comment by cfzjywxk** [comment]
```
go: updates to go.mod needed; to update it:
	go mod tidy
Error: Process completed with exit code 1.
```
Why the CI keep failing like this?

**Comment by you06** [comment]
> Why the CI keep failing like this? 🤔

Also need to update [`integration_tests/go.mod`](https://github.com/tikv/client-go/blob/master/integration_tests/go.mod).

**Comment by cfzjywxk** [comment]
@you06 
Thanks for the reminding.

**Review by you06 — APPROVED** [review]

**Pull request merged**